File: Code of JAVA/SocketReciveTest/src/com/company/ConnectThread.java
package com.company;

import com.alibaba.fastjson.JSONObject;
import com.company.bean.TestPack;
import com.company.bean.UserOrdinary;


import java.beans.ParameterDescriptor;
import java.io.*;
import java.net.Socket;
import java.nio.charset.StandardCharsets;


public class ConnectThread extends Thread{
	BufferedReader input;
	private BufferedWriter out = null;
	private Socket sc;

	static JSONObject jsonData = null;

	String message;

	public ConnectThread(Socket sc) {
		this.sc = sc;
	}
	
	@Override
	public void run()  {
		try {
			System.out.println("thread start");
			input = new BufferedReader(new InputStreamReader(sc.getInputStream(), StandardCharsets.UTF_8));
			out = new BufferedWriter(new OutputStreamWriter(sc.getOutputStream(), StandardCharsets.UTF_8));
			message = input.readLine();		//从连接中读入一行
			if(message == null){
				System.out.println("error:null cilent message ");
			}
			System.out.println("recived cilent message:"+message);
		} catch (Exception e) {
			e.printStackTrace();
		}
		String type = getDisposeType(message);		//解析操作类型
		System.out.println("get message type:"+type);
		disposeOperation(type);		//根据类型进行操作
	}


	/**
	 * 获取客户端的操作类型
	 * @param message 客户端发送的字符串消息
	 * @return	操作类型
	 */
	public String getDisposeType(String message)  {
			//在任何操作前都要记得判断对象非空
			if(message == null || message.equals("")){
				return null;
			}
			jsonData = JSONObject.parseObject(message);		//将jsonstring转换为jsonobject
			String type;
			type = String.valueOf(jsonData.get("type"));
			return type;
	}

	/**
	 * 根据操作类型进行操作
	 * @param type	操作类型
	 */
	private void disposeOperation(String type) {
		if(type != null) {
			switch (type) {
			case "search":{
				if(jsonData.get("login") != null) {
				String id = "",password = "";
				id = (String)jsonData.get("id");
				System.out.println("get id:"+id);
				password = (String)jsonData.get("password");
				System.out.println("get password:"+ password);
				login(id,password);
				}
				if(jsonData.get("getBookInfo")!=null){
					String bookId = (String)jsonData.get("bookNum");
				}
				break;
			}
			case "updata":{
				System.out.println("get updata type，but no method find");
				break;
			}
			default:{
				System.out.println("none type get!");
				break;
			}
			}	
		}
	}

	/**
	 * 登陆操作
	 * @param id	用户id
	 * @param password	用户密码
	 */
	private void login(String id,String password) {
		String sendData = "";
		System.out.println("login successful");
		UserOrdinary testpack = new UserOrdinary();	//要发回的用户消息
		/*
		这两个非空判断只是用于简单测试，实际使用中应该先是在数据库中查询有无对应用户，有的话就查询该用户信息，然后将用户信息打包发回客户端
		 */
		if(id != null){
			testpack.setId(Integer.parseInt(id));
		}
		if(password != null){
			testpack.setPassword(password);
		}
		testpack.setAccountNum("0000000");
		JSONObject jsonUser = (JSONObject)JSONObject.toJSON(testpack);	//javabean转换为jsonstring
		jsonUser.put("type","search");
		jsonUser.put("login","1");
		System.out.println("user data："+jsonUser.toJSONString());
		sendData = jsonUser.toJSONString();
		try {
			out.write(sendData+"\n");
			out.flush();
		} catch (IOException e) {
			e.printStackTrace();
		}
		System.out.println("send user data successful!");


	}

}

File: Code of Kotlin/SocketReciveTest/src/com/company/ConnectThread.kt
package com.company

import com.alibaba.fastjson.JSON
import com.alibaba.fastjson.JSONObject
import com.company.bean.UserOrdinary
import java.io.*
import java.net.Socket
import java.nio.charset.StandardCharsets
import kotlin.reflect.jvm.internal.impl.load.kotlin.JvmType

class ConnectThread(private val sc: Socket) : Thread() {
    var input: BufferedReader? = null
    var out: BufferedWriter? = null
    var message: String? = null
    var reply:String? = null
    override fun run() {
        val ct:Connect = Connect(sc)
        ct.InitConnect()
        try {
            doLog.addLog("thread start:"+this.name)
            doLog.addLog("客户端ip地址是："+sc.getInetAddress());
            message = ct.receiveMessage()
            if (message == null) {
                doLog.addLog("error:null cilent message ","error")
                return
            }
            doLog.addLog("recived cilent message:$message")
        } catch (e: Exception) {
            e.printStackTrace()
            doLog.addLog(e.toString())
        }
        val type = getDisposeType(message)
        if(type != null){
            doLog.addLog("get message type:$type")
            disposeOperation(type)
        }else{
            when(message){
                "2333","1","ez" ->{
                    reply = "Connected!"
                }
                else ->{

                }
            }
        }
        ct.sendMessage(reply)
        ct.CloseConnect()
    }

    fun getDisposeType(message: String?): String? {
        if (message == null || message == "" || message[0] != '{') {
            return null
        }
        jsonData = JSONObject.parseObject(message)
        val type: String = jsonData!!.get("type").toString()
        return type
    }

    private fun disposeOperation(type: String?) {
        if (type != null) {
            when (type) {
                "search" -> {
                    if (jsonData!!["login"] != null) {
                        var id: String? = ""
                        var password: String? = ""
                        id = jsonData!!["id"] as String?
                        doLog.addLog("get id:$id")
                        password = jsonData!!["password"] as String?
                        doLog.addLog("get password:$password")
                        login(id, password)
                    }
                }
                "updata" -> {
                    doLog.addLog("get updata type，but no method find")
                }
                else -> {
                    doLog.addLog("none type get!")
                }
            }
        }
    }

    private fun login(id: String?, password: String?) {
        var sendData = ""
        doLog.addLog("login successful")
        val testpack = UserOrdinary()
        if (id != null) {
            testpack.id = id.toInt()
        }
        if (password != null) {
            testpack.password = password
        }
        testpack.accountNum = "0000000"
        val jsonUser = JSONObject.toJSON(testpack) as JSONObject
        jsonUser["type"] = "search"
        jsonUser["login"] = "1"
        doLog.addLog("user data：" + jsonUser.toJSONString())
        sendData = jsonUser.toJSONString()
        try {
            out!!.write("""
    $sendData
    
    """.trimIndent())
            out!!.flush()
        } catch (e: IOException) {
            e.printStackTrace()
            doLog.addLog(e.toString())

        }
        doLog.addLog("send user data successful!")
    }

    companion object {
        var jsonData: JSONObject? = null
    }

}